client/my-sites/domains/domain-search/index.jsx

```diff
@@ -155,6 +156,11 @@ class DomainSearch extends Component {
 				fillInSingleCartItemAttributes( registration, this.props.productsList ),
 			] )
 			.then( () => {
+				if ( config.isEnabled( 'titan/upsell-email-domain-purchase' ) ) {
+					page( domainAddEmailUpsell( this.props.selectedSiteSlug, domain ) );
+					return;
+				}
+
 				if ( this.props.userCanPurchaseGSuite && canDomainAddGSuite( domain ) ) {
```

Comment:
@daledupreez you mentioned removing GSuite-specific logic on the Asana task but what needs to be done about this particular condition? Should this condition be moved into the `<EmailProvidersComparison>` component and the GSuite card conditionally rendered based on its truthiness? 🤔

Reply:
This logic is already embedded in `EmailProvidersComparison`:
https://github.com/Automattic/wp-calypso/blob/94c00e90e144208f73488287905df71e95e63dad/client/my-sites/email/email-providers-comparison/index.jsx#L694

However, I do notice that logic needs to be modified slightly due to us now not having a `domain` object coming from this upsell path. I think that code needs to be updated to the following:
```javascript
const isGSuiteSupported = canUserPurchaseGSuite( state ) && ( hasCartDomain || ( domain && hasGSuiteSupportedDomain( [ domain ] ) ) ); 
```

Reply:
Thanks for the feedback! I'll test this out.

Reply:
I've made the change and tested it by returning `true`/`false`.